zing_stats.py

```diff
@@ -164,140 +180,212 @@ def main():
+            teams_map['All'].append(name)
+        if name not in teams_map[team]:
+            teams_map[team].append(name)
+    log.debug('teams map: %s', teams_map)
 
     if args.range_hours <= 24:
         file_prefix = 'last_%dh' % args.range_hours
     else:
         file_prefix = 'last_%gd' % round((args.range_hours / 24), 1)
-
-    if args.report_format == 'html':
-        write_html(args, df, file_prefix, len(changes), start_dt, finish_dt,
-                   projects, proj_map)
-
-    elif args.report_format == 'json':
-        write_json(args, df, file_prefix)
-
-
-def write_html(args, df, file_prefix, num_changes, start_dt, finish_dt,
-               all_projects, proj_map):
-    if proj_map:
-        # convert project maps to groups into group map to projects
-        group_map = dict()
-        for group in set(proj_map.values()):
-            group_map[group] = list()
-            for project in proj_map:
-                if proj_map[project] == group:
-                    group_map[group].append(project)
-
-        # and add an all group for all projects
-        group_map['All'] = proj_map.keys()
-        log.debug('group_map: %s', group_map)
-
-        for group in group_map:
-            projects = group_map[group]
-            groups = group_map.keys()
-            html = generate_html(args, df, num_changes, start_dt, finish_dt,
-                                 projects, all_projects, group, groups)
-            dir_path = os.path.join(args.output_dir, file_prefix)
-            if group == 'All':
-                file_name = 'index.html'
-            else:
-                file_name = '%s.html' % group.lower()
-            file_path = os.path.join(dir_path, file_name)
-            if not os.path.exists(dir_path):
-                os.makedirs(dir_path)
-            with open(file_path, 'w') as f:
-                f.write(html)
-            log.info('Wrote %s for %s group', file_path, group)
-    else:
+    all_projects = teams_map['All']
+    for team in sorted(teams_map):
+        team_projects = teams_map[team]
+        teams = teams_map.keys()
+        html = generate_html(args, df, num_changes, start_dt, finish_dt,
+                             team_projects, all_projects, team, teams)
         dir_path = os.path.join(args.output_dir, file_prefix)
-        file_name = 'index.html'
+        if team == 'All':
+            file_name = 'index.html'
+        else:
+            file_name = '%s.html' % re.sub(r'\W+', '_', team.lower())
         file_path = os.path.join(dir_path, file_name)
         if not os.path.exists(dir_path):
             os.makedirs(dir_path)
-        html = generate_html(args, df, num_changes, start_dt, finish_dt,
-                             df.keys(), all_projects, 'All')
         with open(file_path, 'w') as f:
             f.write(html)
-        log.info('Wrote %s', file_path)
+        log.info('Wrote %s for team: "%s"', file_path, team)
+
+
+def gather_github_prs(args, oldest_timestamp, projects):
```

Comment:
organizationally, consider moving the code hosting specific parts to separate files within the project, that way it would be easier for someone to add additional sites such as bitbucket and gitlab by simply implementing the needed methods and entrypoints to help call.

Reply:
Thanks, I agree this would be useful - I need to think about how to restructure this to make support for new systems be more pluggable. This is very much a first implementation at adding a second system and still needs cleanup. But I think whats here is useful enough to land now and iterate on.